enterprise_access/apps/content_assignments/api.py

```diff
@@ -560,14 +581,19 @@ def _create_new_assignments(
 
     # First, prepare assignment objects using data available in-memory only.
     content_title = _get_content_title(assignment_configuration, content_key)
+    parent_content_key = _get_parent_content_key(assignment_configuration, content_key)
     preferred_course_run_key = _get_preferred_course_run_key(assignment_configuration, content_key)
+    is_assigned_course_run = bool(parent_content_key)
```

Comment:
[inform] Uses a non-null `parent_content_key` to determine whether the assignment is run-based.